pytorch_lightning/callbacks/stochastic_weight_avg.py

```diff
@@ -179,7 +179,8 @@ def on_train_epoch_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningMo
                 anneal_strategy=self._annealing_strategy,
                 last_epoch=trainer.max_epochs if self._annealing_strategy == "cos" else -1,
             )
-            default_scheduler_cfg = LRSchedulerConfig(self._swa_scheduler)
+            # We assert that there is only one optimizer on fit start, so know opt_idx is always 0
+            default_scheduler_cfg = LRSchedulerConfig(self._swa_scheduler, opt_idx=0)
```

Comment:
Alternatively, maybe `LRSchedulerConfig.opt_idx` should default to 0 instead of None?

Reply:
yeah! I think that'd be better since we always assume each lr scheduler config should have an opt_idx.

Reply:
yeah! I think that'd be better

Reply:
We use the `None` default to know if the user set it manually:

https://github.com/PyTorchLightning/pytorch-lightning/blob/28dac0c8d9bba4331793c932694eda4eab0bb94e/pytorch_lightning/core/optimizer.py#L361

Reply:
oh yeah!! forgot about this. makes sense :)